### ISSUE TITLE
Panic with guestAccelerators for NodePool

### ISSUE THREAD

**Issue by ledniy** [pulumi/pulumi-gcp#1435]
### What happened?

Whenever I add guestAccelerators to a NoodPool I'm getting panic on `pulumi up` and even with `--skip-preview --yes`
```
Previewing update (global):
     Type                       Name                         Plan       Info
     pulumi:pulumi:Stack        k8s-cluster-global             1 error; 40 messages
 +   └─ gcp:container:NodePool  gpu-node-pool                create     1 error

Diagnostics:
  pulumi:pulumi:Stack (k8s-cluster-global):
    panic: value has no attribute of that name
    goroutine 104 [running]:
    github.com/hashicorp/go-cty/cty.Value.GetAttr({{{0x1058ed6b8?, 0x14007928760?}}, {0x1050cbee0?, 0x14007c007b0?}}, {0x104055af3, 0x7})
    	/home/runner/go/pkg/mod/github.com/hashicorp/go-cty@v1.4.1-0.20200414143053-d3edf31b6320/cty/value_ops.go:700 +0x26c
    github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource.GetProjectFromDiff(0x1400691ed88?, 0x14003c64000)
    	/home/runner/work/pulumi-gcp/pulumi-gcp/upstream/google-beta/tpgresource/utils.go:110 +0x84
    github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource.DefaultProviderProject({0x14007ca7aa0?, 0x14007ca79b0?}, 0x1400705af30?, {0x105861a60?, 0x14003c64000})
    	/home/runner/work/pulumi-gcp/pulumi-gcp/upstream/google-beta/tpgresource/utils.go:762 +0x68
    github.com/hashicorp/terraform-provider-google-beta/google-beta/services/container.ResourceContainerNodePool.All.func1({0x1058ed108, 0x10967fae0}, 0x10406a477?, {0x105861a60, 0x14003c64000})
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20230912190043-e6d96b3b8f7e/helper/customdiff/compose.go:53 +0x98
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0x140013c8ab0, {0x1058ed108, 0x10967fae0}, 0x140011c9110, 0x14007c00630, 0x14000b017c0, {0x105861a60, 0x14003c64000}, 0x0)
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20230912190043-e6d96b3b8f7e/helper/schema/schema.go:698 +0x3b8
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0x0?, {0x1058ed108?, 0x10967fae0?}, 0x140011c9110, 0x140011c8dd0?, {0x105861a60?, 0x14003c64000?})
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20230912190043-e6d96b3b8f7e/helper/schema/resource.go:967 +0x9c
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.simpleDiff({0x1050ce1a0?, {0x1400146e720?, 0x1040aff50?, 0x1a?}}, 0x140013c8ab0?, 0x100a5ad9c?, 0x104e36f00?, 0x14007c00360?, {{{0x1058ed6b8?, 0x14007928760?}}, ...}, ...)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.67.1-0.20231208194349-bef42c24d09a/pkg/tfshim/sdk-v2/provider_diff.go:101 +0x808
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.Diff({0x140000cff20?, {0x0?, 0x90?, 0x1400691f5a8?}}, {0x1040aff50, 0x1a}, {0x0?, 0x0}, {0x1058bcd20, 0x14007c00630})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.67.1-0.20231208194349-bef42c24d09a/pkg/tfshim/sdk-v2/provider_diff.go:63 +0x1fc
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Create(0x140005c9980, {0x1058ed028?, 0x1400702a540?}, 0x140070300f0)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.67.1-0.20231208194349-bef42c24d09a/pkg/tfbridge/provider.go:842 +0x468
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Create.func1({0x105905ce8?, 0x140005c9980?})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:344 +0x3c
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.resourceMethod[...](0x14006473220?, 0x1400691f778, 0x1400691f758?)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:303 +0xc4
    github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Create(0x1400691f7b8?, {0x1058ed028?, 0x1400702a540?}, 0x104eb02c0?)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:343 +0x5c
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler.func1({0x1058ed028, 0x1400702a540}, {0x105507480?, 0x140070300f0})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.95.0/proto/go/provider_grpc.pb.go:593 +0x74
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x1058ed028, 0x1400702a180}, {0x105507480, 0x140070300f0}, 0x14007038080, 0x14007026168)
    	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x2e8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler({0x1058157a0?, 0x14006473220}, {0x1058ed028, 0x1400702a180}, 0x1400703a000, 0x1400247c6a0)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.95.0/proto/go/provider_grpc.pb.go:595 +0x12c
    google.golang.org/grpc.(*Server).processUnaryRPC(0x14000036780, {0x105900720, 0x140055e2340}, 0x14007036000, 0x140064c3920, 0x10962c900, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
    google.golang.org/grpc.(*Server).handleStream(0x14000036780, {0x105900720, 0x140055e2340}, 0x14007036000, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:1751 +0x7c4
    google.golang.org/grpc.(*Server).serveStreams.func1.1()
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
    created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 44
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:997 +0x160

    error: preview failed

  gcp:container:NodePool (gpu-node-pool):
    error: Preview failed: error reading from server: EOF


```

### Example

```
guestAccelerators: [{
  type: "nvidia-tesla-t4",
  count: 1
}]
```

### Output of `pulumi about`

CLI
Version      3.96.2
Go Version   go1.21.4
Go Compiler  gc

Plugins
NAME    VERSION
gcp     7.2.2
nodejs  unknown

Host
OS       darwin
Version  14.0
Arch     arm64

This project is written in nodejs: executable='/Users/ledniy/.nvm/versions/node/v20.5.0/bin/node' version='v20.5.0'

Current Stack: organization/k8s-cluster/global

TYPE                               URN
pulumi:pulumi:Stack                urn:pulumi:global::k8s-cluster::pulumi:pulumi:Stack::k8s-cluster-global
pulumi:providers:gcp               urn:pulumi:global::k8s-cluster::pulumi:providers:gcp::default_7_2_2
gcp:compute/network:Network        urn:pulumi:global::k8s-cluster::gcp:compute/network:Network::network
gcp:compute/subnetwork:Subnetwork  urn:pulumi:global::k8s-cluster::gcp:compute/subnetwork:Subnetwork::subnet
gcp:container/cluster:Cluster      urn:pulumi:global::k8s-cluster::gcp:container/cluster:Cluster::main

Found no pending operations associated with global

Backend
Name           ledniy.local
URL            file://~
User           ledniy
Organizations
Token type     personal

Dependencies:
NAME            VERSION
@types/node     18.19.3
typescript      4.9.5
@pulumi/gcp     7.2.2
@pulumi/pulumi  3.97.0

Pulumi locates its logs in /var/folders/jn/8yt07zxd0957nt01h7z6fnym0000gn/T/ by default

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by t0yv0** [comment]
Thank you for reporting this! Our team will get on this shortly.

**Comment by guineveresaenger** [comment]
@ledniy - it would help us diagnose if you could send us a minimal working example that reproduces this bug. Thank you so much!

**Comment by ledniy** [comment]
Here's what I did:
1. `pulumi new gcp-typescript`
2. Added gpuNodePool with following
```
const gpuNodePool = new gcp.container.NodePool("gpu-node-pool", {
    cluster: gkeCluster.id,
    initialNodeCount: 1,
    nodeConfig: {
        machineType: "n1-highmem-8",
        diskSizeGb: 50,
        oauthScopes: ["https://www.googleapis.com/auth/cloud-platform"],
        guestAccelerators: [{
            type: "nvidia-tesla-t4",
            count: 1
        }]
    }
});
```

**Comment by VenelinMartinov** [comment]
Here is a full program which exhibits this error:
```typescript
import * as pulumi from "@pulumi/pulumi";
import * as gcp from "@pulumi/gcp";

// Create a GKE cluster
const engineVersion = gcp.container.getEngineVersions().then(v => v.latestMasterVersion);
const cluster = new gcp.container.Cluster("cluster", {
    initialNodeCount: 2,
    minMasterVersion: engineVersion,
    nodeVersion: engineVersion,
    nodeConfig: {
        machineType: "n1-standard-1",
        oauthScopes: [
            "https://www.googleapis.com/auth/compute",
            "https://www.googleapis.com/auth/devstorage.read_only",
            "https://www.googleapis.com/auth/logging.write",
            "https://www.googleapis.com/auth/monitoring"
        ],
    },
});

const gpuNodePool = new gcp.container.NodePool("gpu-node-pool", {
    cluster: cluster.id,
    initialNodeCount: 1,
    nodeConfig: {
        machineType: "n1-highmem-8",
        diskSizeGb: 50,
        oauthScopes: ["https://www.googleapis.com/auth/cloud-platform"],
        guestAccelerators: [{
            type: "nvidia-tesla-t4",
            count: 1
        }]
    }
});
```

running `pulumi preview` on this yields the panic.

Commenting out the `guestAccelerators` property prevents the panic. I also tested v6 and it doesn't seem to panic either.

**Comment by VenelinMartinov** [comment]
Upon digging some more, this seems like it might be related to https://github.com/pulumi/pulumi-terraform-bridge/issues/911

I tested overriding `opts.diffStrategy = PlanState` in the test and the panic goes away. I'll dig some more here.

**Comment by t0yv0** [comment]
Thank you for researching this!

Couple of things, let's first upgrade to the latest bridge version. If the schema calls for a list but a tuple is produced this indicates an issue with type-aware translation of data to the cty.Value domain. makeResourceRawConfigClassic is the legacy method that uses destination-type-unaware method by just inspecting incoming json-like data.  

Recently the improved method received some more fixes:
https://github.com/pulumi/pulumi-terraform-bridge/pull/1568

And here is a plan we had for rolling out opts.diffStrategy = PlanState which we have done already for pulumi-aws.
https://github.com/pulumi/pulumi-terraform-bridge/issues/866

Given this context I'd be supportive of opting GCP into opts.diffStrategy = PlanState (and upgrading to the latest bridge) especially if this fixes the bug, as it probably will fix more issues as well.

**Comment by VenelinMartinov** [comment]
We merged a fix for this and released it in [7.3.1](https://github.com/pulumi/pulumi-gcp/releases/tag/v7.3.1).
